RocketChat/Rocket.Chat.ReactNative
# [FIX] Messages attributed to wrong user

## Proposed changes


Under certain circumstances, messages sent by the app user are shown as originating from another user.

## Issue(s)	


Closes https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4559

## How to test or reproduce


You need two different users.

- In a desktop browser, open [Rocket.Chat](http://rocket.chat/) and login with one of the users. Navigate to an arbitrary channel.

- Open the [Rocket.Chat](http://rocket.chat/) app and login with the other user. Navigate to the same channel as in step 1.

- Submit a message from the app.

- Prepare another message in the app, but do not submit it right now.

- Send a message from the desktop browser.

- Submit the message prepared in step 4. It is important to submit the message before the message from step 5 is displayed in the app.

- Submit another message from the app.

## Screenshots

#### Before

https://user-images.githubusercontent.com/47038980/194583134-03c87211-b570-4a07-ba15-84a11d9844ba.mov

#### After


https://user-images.githubusercontent.com/47038980/194583165-20193fcb-614b-41c4-beb1-b3777dfd399f.mov



## Types of changes




## Checklist



## Further comments


Jira: [TC-160]

[TC-160]: https://rocketchat.atlassian.net/browse/TC-160?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
